Whiskey/Tasks/File.ps1

```diff
@@ -0,0 +1,50 @@
+function New-WhiskeyFile
+{
+    [Whiskey.Task('File')]
+    [CmdletBinding()]
+    param(
+        [Whiskey.Tasks.ValidatePath(Mandatory,PathType='File',MustExist=$false)]
+        [string[]]$Path,
+
+        [Parameter(Mandatory)]
+        [Whiskey.Context]$TaskContext,
+
+        [string]$Content,
+
+        [switch]$Touch
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
+
+    foreach( $item in $Path )
+    {
+        if( Test-Path -Path (Split-Path -Path $item) -PathType Leaf )

```

Comment:
I think this can be simplified. Don't check if the parent is a file or if the item is a directory. There are a million weird situations we could test for. Let `New-Item` fail if something is weird.

```
if( -not (Test-Path -Path $item -ItemType Leaf) )
{
     New-Item -Path $item -ItemType 'File' -Force -ErrorAction Stop
}

if( $Touch )
{
}

if( $Content )
{
}
```

Reply:
Should I be keeping these as "elseif" statements or should I change them all to "if"? Because only one of these conditions need to trigger for each item right?

Reply:
I guess it wouldn't cause any problems now that I am thinking about it, but it might set content twice, for example.

Reply:
Standalone `if` statements. If you don't pass `Value` to the `New-Item` cmdlet, you'll only set content once.